include/seqan3/io/alignment_file/output.hpp

```diff
@@ -0,0 +1,706 @@
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \brief Provides seqan3::alignment_file_output and corresponding traits classes.
+ * \author Svenja Mehringer <avenja.mehringer AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <cassert>
+#include <fstream>
+#include <string>
+#include <string_view>
+#include <variant>
+#include <vector>
+
+#include <seqan3/core/concept/tuple.hpp>
+#include <seqan3/core/metafunction/basic.hpp>
+#include <seqan3/core/metafunction/template_inspection.hpp>
+
+#include <seqan3/io/alignment_file/format_sam.hpp>
+#include <seqan3/io/alignment_file/header.hpp>
+#include <seqan3/io/alignment_file/output_format_concept.hpp>
+#include <seqan3/io/alignment_file/output_options.hpp>
+#include <seqan3/io/detail/out_file_iterator.hpp>
+#include <seqan3/io/detail/record.hpp>
+#include <seqan3/io/exception.hpp>
+#include <seqan3/io/filesystem.hpp>
+#include <seqan3/io/record.hpp>
+#include <seqan3/std/ranges>
+
+namespace seqan3
+{
+
+// ----------------------------------------------------------------------------
+// alignment_file_output
+// ----------------------------------------------------------------------------
+
+/*!\brief A class for writing alignment files, e.g. SAM, BAL, BLAST, ...
+ * \ingroup alignment_io
+ * \tparam selected_field_ids   A seqan3::fields type with the list and order of
+ *                              fields IDs; only relevant if these can't be deduced.
+ * \tparam valid_formats        A seqan3::type_list of the selectable formats (each
+ *                              must model seqan3::alignment_file_output_format_concept).
+ * \tparam stream_type          The type of the stream, must model seqan3::ostream_concept.
+ *
+ * \details
+ *
+ * ### Introduction
+ *
+ * Alignment files provide a way to store pairwise alignment information.
+ * Well-known formats include SAM and BAM or BLAST.
+ *
+ * The Alignment file abstraction supports writing following fields:
+ *
+ *   1. field::SEQ
+ *   2. field::ID
+ *   3. field::OFFSET
+ *   4. field::REF_SEQ
+ *   5. field::REF_ID
+ *   6. field::REF_OFFSET
+ *   7. field::ALIGNMENT
+ *   8. field::MAPQ
+ *   9. field::FLAG
+ *   10. field::QUAL
+ *   11. field::MATE
+ *   12. field::TAGS
+ *   13. field::EVALUE
+ *   14. field::BIT_SCORE
+ *
+ * There is an additional field called `field::HEADER_PTR`, it used to transfer
+ * header information from `alignment_file_input` to `alignment_file_output`,
```

Comment:
it **is** used

no backticks around the stuff, but instead `seqan3::` before the files. always include namespace to make sure that links are generated :point_up: